src/eager.test.ts

```diff
@@ -0,0 +1,61 @@
+import { pipe } from './pipe'
+import { eager } from './eager'
+import { map } from './map'
+import { wait } from './delay'
+import { toArray } from './toArray'
+
+async function bench(cb: Function) {
+  const start = process.hrtime.bigint()
+  await cb()
+  const end = process.hrtime.bigint()
+  return Number((end - start) / BigInt(1e6)) // Nanoseconds to milliseconds
+}
+
+it('should be possible to eager load an API call for example (from sync -> async)', async () => {
+  function fetch(path: string) {
+    return new Promise(resolve => setTimeout(resolve, 50, path))
+  }
+
+  const program = pipe(
+    map(id => `/user/${id}`),
+    map<string, unknown>(path => fetch(path)),
+    eager(10),
```

Comment:
If this test is processing 10 items, what about testing in batches of `5` or `4`, just to spice it up a bit? 😄

Reply:
Oh yes I agree. This allows me to implicitly test that it is doing the batching correctly!